bitvector32.go

```diff
@@ -0,0 +1,88 @@
+package bitfield
+
+import (
+	"math/bits"
+)
+
+var _ = Bitfield(Bitvector32{})
+
+// Bitvector32 is a bitfield with a fixed defined size of 32. There is no length bit
+// present in the underlying byte array.
+type Bitvector32 []byte
+
+// NewBitvector32 creates a new bitvector of size 32.
+func NewBitvector32() Bitvector32 {
+	byteArray := [8]byte{}
+	return byteArray[:]
+}
+
+// BitAt returns the bit value at the given index. If the index requested
+// exceeds the number of bits in the bitvector, then this method returns false.
+func (b Bitvector32) BitAt(idx uint64) bool {
+	// Out of bounds, must be false.
+	if idx >= b.Len() {
+		return false
+	}
+
+	i := uint8(1 << (idx % 8))
+	return b[idx/8]&i == i
+}
+
+// SetBitAt will set the bit at the given index to the given value. If the index
+// requested exceeds the number of bits in the bitvector, then this method returns
+// false.
+func (b Bitvector32) SetBitAt(idx uint64, val bool) {
+	// Out of bounds, do nothing.
+	if idx >= b.Len() {
+		return
+	}
+
+	bit := uint8(1 << (idx % 8))
+	if val {
+		b[idx/8] |= bit
+	} else {
+		b[idx/8] &^= bit
+	}
+}
+
+// Len returns the number of bits in the bitvector.
+func (b Bitvector32) Len() uint64 {
+	return uint64(len(b) * 8)
+}
```

Comment:
Len returns the number of bits in the bitvector? That doesn't seem to match the implementation. Wouldn't it always be 32?